packages/proxy-agent/src/index.ts

```diff
@@ -99,15 +97,38 @@ export class ProxyAgent extends Agent {
 		this.getProxyForUrl = opts?.getProxyForUrl || envGetProxyForUrl;
 	}
 
-	async connect(
-		req: http.ClientRequest,
-		opts: AgentConnectOpts
-	): Promise<http.Agent> {
-		const { secureEndpoint } = opts;
+	private getProxyData(req: http.ClientRequest, opts: AgentConnectOpts) {
+		const secureEndpoint = opts.secureEndpoint ?? this.isSecureEndpoint();
```

Comment:
In the "proxy-agent" package, I had to change the `secureEndpoint` option to fall back to `this.isSecureEndpoint()` on the base class.

The reason is that [the `secureEndpoint` option is only set in "agent-base" when `Agent.createSocket()` called](https://github.com/TooTallNate/proxy-agents/blob/2f835a41f265280192b82556db80ccbe67140753/packages/agent-base/src/index.ts#L56-L66) (this means that the first time `Agent.addRequest()` is called, the `secureEndpoint` option is present, but in all subsequent requests, the `secureEndpoint` option is missing).

Reply:
I removed this change in bac30180bd8e629dba7393968d05786ceede0768 in order to descope this PR